src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -271,6 +274,13 @@ public ExamScoresDTO calculateExamScores(Long examId) {
 
             if (scores.maxPoints != null) {
                 studentResult.overallScoreAchieved = (studentResult.overallPointsAchieved / scores.maxPoints) * 100.0;
+                // Sets grading scale related properties for exam scores
+                Optional<GradingScale> gradingScale = gradingScaleRepository.findByExamId(examId);
+                if (gradingScale.isPresent()) {
+                    GradeStep studentGrade = gradingScaleService.matchPercentageToGradeStep(studentResult.overallScoreAchieved, gradingScale.get().getId());
```

Comment:
can you please move `matchPercentageToGradeStep` into a repository?
Then you can avoid the dependency to the `gradingScaleService` above.

We should in general move logic into the repository if possible to avoid too many dependencies between services.
This is always possible if the method involves only one repository, then you can easily move the method into this repository